ts-sdk/whirlpool/src/createPool.ts

```diff
@@ -0,0 +1,203 @@
+import {
+  getFeeTierAddress,
+  getInitializePoolV2Instruction,
+  getInitializeTickArrayInstruction,
+  getTickArrayAddress,
+  getTickArraySize,
+  getTokenBadgeAddress,
+  getWhirlpoolAddress,
+  getWhirlpoolSize,
+} from "@orca-so/whirlpools-client";
+import type {
+  Address,
+  GetMinimumBalanceForRentExemptionApi,
+  GetMultipleAccountsApi,
+  IInstruction,
+  LamportsUnsafeBeyond2Pow53Minus1,
+  Rpc,
+  TransactionPartialSigner,
+} from "@solana/web3.js";
+import { generateKeyPairSigner } from "@solana/web3.js";
+import {
+  DEFAULT_ADDRESS,
+  DEFAULT_FUNDER,
+  SPLASH_POOL_TICK_SPACING,
+  WHIRLPOOLS_CONFIG_ADDRESS,
+} from "./config";
+import invariant from "tiny-invariant";
+import {
+  getFullRangeTickIndexes,
+  getTickArrayStartTickIndex,
+  priceToSqrtPrice,
+  sqrtPriceToTickIndex,
+} from "@orca-so/whirlpools-core";
+import { fetchAllMint, getTokenSize } from "@solana-program/token";
+
+type CreatePoolInstructions = {
+  instructions: IInstruction[];
+  initializationCost: LamportsUnsafeBeyond2Pow53Minus1;
+  poolAddress: Address;
+};
+
+export function createSplashPoolInstructions(
+  rpc: Rpc<GetMultipleAccountsApi & GetMinimumBalanceForRentExemptionApi>,
+  tokenMintOne: Address,
+  tokenMintTwo: Address,
+  initialPrice: number = 1,
+  funder: TransactionPartialSigner = DEFAULT_FUNDER,
+): Promise<CreatePoolInstructions> {
+  return createConcentratedLiquidityPoolInstructions(
+    rpc,
+    tokenMintOne,
+    tokenMintTwo,
+    SPLASH_POOL_TICK_SPACING,
+    initialPrice,
+    funder,
+  );
+}
+
+export async function createConcentratedLiquidityPoolInstructions(
+  rpc: Rpc<GetMultipleAccountsApi & GetMinimumBalanceForRentExemptionApi>,
+  tokenMintOne: Address,
+  tokenMintTwo: Address,
+  tickSpacing: number,
+  initialPrice: number = 1,
```

Comment:
here, is number (f64) used for non-display purpose ?

Reply:
Yeah I think here its a tradeoff between:
* User is able to specify the initial price in something meaningful (i.e. actual price is a lot more meaningful than a sqrt_price).
* User is able to specify the exact price they want (impossible with actual price since there might be some rounding error).

I think in this instance the initialSqrtPrice doesn't have to be exact. e.g. is someone really going to care that their pool got initialized with a price of 100.00000 vs 100.00001?